app/routes/$repo.tsx

```diff
@@ -49,12 +50,39 @@ export default function Repo() {
       contentDisplayName={content.displayName}
       contentPath={content.contentName}
       sidebarConfig={content.schema?.sidebar}
+      internalSidebarMenu={{
+        selectedTool: contentToolMap[content.contentName],
+        toolLinks: toolLinks,
+      }}
     >
       <Outlet />
     </InternalPage>
   )
 }
 
+const toolContentMap: Record<ToolName, ContentName> = {
+  cadence: "cadence",
+  cli: "flow-cli",
+  emulator: "flow-emulator",
+  fcl: "fcl-js",
+  testing: "flow-js-testing",
+  vscode: "vscode-extension",
+}
+
+const toolLinks: Record<ToolName, string> = { ...toolContentMap }
+for (let [key, value] of Object.entries(toolLinks)) {
+  toolLinks[key as ToolName] = `/${value}`
+}
+
+const contentToolMap: Record<string, ToolName> = {
```

Comment:
Are these just reversed mappings of each other? Are they meant to diverge at any point? Could be prone to errors, or updating one and not the other.  Since `toolContentMap` seems to only be used for generating the `toolLinks`, would it be simpler to just generate that from `contentToolMap` so there's only one place to specify the mapping? Something like this maybe?

```ts
const toolLinks = Object.entries(contentToolMap).map(([content, tool]) => ({ [tool]: `/${content}` }))
```

Reply:
agree that would be better, but IMO the core issue is that this mapping needs to exist at all so i don't think i'm going to bother with the small improvements. IMO al l the tools (and their images etc) should be passed in as props